technology-considerations/Considerations for authentication and authorization technology

```diff
@@ -0,0 +1,25 @@
+| Other   dept usage                        | Own bespoke build                                                | 1     | Digi services in use with   limitations                      | 2     | Not aware of use                                               | 3     | This is also used in other gov   depts so I've raised Auth0 to 3 and set keycloak to be the same                                        | 3     |
+| Failure   impact of component             | Risks, but own build to fix                                      | 2     | Risks, but more known                                        | 2     | Risks, unknown                                                 | 1     | We have full control over the   upgrade cycle of the software and the hosting. They are our risks but we have   full control over them. | 3     |
+| Security   risk mitigation                | Unknown, but should be able to   build with time                 | 3     | Known token issue threat                                     | 4     | Seems most secure                                              | 5     | As secure as auth0, it is an   OIDC provider                                                                                            | 4     |
+| Supports   GDPR                           | We need work to prove                                            | 3     | Established                                                  | 5     | Established                                                    | 5     | We need work to prove                                                                                                                   | 3     |
+| Supports   GDS                            | On balance                                                       | 3     | On balance                                                   | 3     | On balance                                                     | 3     | On balance                                                                                                                              | 3     |
+| Supports   WCAG                           | Can build what's needed with   time                              | 5     | Lack of customisation                                        | 3     | Limited customisation                                          | 3     | Complete customisation including   existing themes for the GDS design system                                                            | 5     |
+| Total   Scores                            | Silhouette with own                                              | 61    | Cognito                                                      | 66    | 0Auth                                                          | 63    | Keycloak                                                                                                                                | 68    |
```

Comment:
Could we add a line to say that we were just using the scores for information, not as the definitive decision?

I think it would also be helpful for us when we're looking back at this if we added a few lines summarising the main factors that ended up influencing the decision.

Reply:
Added a few lines.